src/utils/hooks/useFetch.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useRef, useState } from "react";
+import { useApiRef } from "../../store/ApiRefContext";
+import { objectIsEmpty } from "../helpers";
+
+const useFetch = function (url) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [data, setData] = useState({});
+  const [error, setError] = useState(null);
+  const cache = useRef({});
+  const { apiRef, isApiRefLoading } = useApiRef();
+
+  useEffect(() => {
+    if (isApiRefLoading) {
+      return () => {};
+    }
+
+    let mounted = true;
+    const abortController = new AbortController();
+
+    async function fetchData() {
+      try {
+        let data = cache.current[url] ?? {}; // First, try to get data from the cache
+        setIsLoading(true);
+
+        // If not found in the cache or skipCache, fetch from the url
+        if (objectIsEmpty(data)) {
```

Comment:
instead of assigning an empty object you can assign null:

```suggestion
        let data = cache.current[url] ?? null; // First, try to get data from the cache
        setIsLoading(true);

        // If not found in the cache or skipCache, fetch from the url
        if (!data) {
```

Reply:
ok